Source/SerializationTransformer.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  SerializationTransformer.swift
+//  Kakapo
+//
+//  Created by Alex Manzella on 27/06/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+/// A closure that given a String as input output the transformed String, used to transform keys of the json
+public typealias KeyTransformer = (key: String) -> String
+
+/**
+ *  A protocol that special `Serializable` objects can adopt to transform other `Serializable` objects.
+ *  At the moment only key transforms are supported, see `SnakecaseTransformer` for a concrete implementations
+ */
```

Comment:
Actually seeing again the sentence, I think this is not actually what this `SerializationTransformer` are being used for I think: right now the objects who use this protocol can only be used to wrap Serializable objects -> by calling serialize on this transformer objects, we will serialize the underlying serializable objects by passing the `KeyTransformer` closure, which will convert the keys.